src/nvim/ui_compositor.c

```diff
@@ -711,11 +711,17 @@ static void ui_comp_event(UI *ui, char *name, Array args)
   bool handled = false;
 
   map_foreach_value(&ui_event_cbs, event_cb, {
+    err.type = kErrorTypeNone;
+    err.msg = NULL;
```

Comment:
It seems that `Error err = ERROR_INIT` declation and `api_clear_error(&err)` should both be moved inside this loop.

Reply:
Sorry for delayed response, Thanksgiving, travelling, learning C macros.

I made an attempt at this and discussed the issue [in the above comment on the PR](https://github.com/neovim/neovim/pull/21147#discussion_r1029735065).

I've made more attempts to get it to work, learning more about C macros along the way. I believe the reason for the `too many arguments provided to function-like macro invocation` compiler error is because the `ERROR_INIT` macro includes a comma:

```C
#define ERROR_INIT { .type = kErrorTypeNone, .msg = NULL }
```

The C preprocessor sees any comma as an argument separator. I could be wrong about this, I gained the insight [from this SO post](https://stackoverflow.com/a/38030161/9168060).

FWIW, defining `ERROR_INIT` with just the `.type`, such as below:

```C
#define ERROR_INIT { .type = kErrorTypeNone }
```

does not trigger a compiler error. Leads me to believe the comma is the issue. I made some attempts at workaround, but couldn't find a solution. I'm curious your opinion.

Reply:
Oh, didn't read that carefully. I guess assigning another variable to `ERROR_INIT` and the assigning `err` to that variable inside the loop should work then.

Reply:
Thank you for the advice! I've followed it, it works.